pydantic/utils.py

```diff
@@ -237,7 +238,7 @@ def in_ipython() -> bool:
     Check whether we're in an ipython environment, including jupyter notebooks.
     """
     try:
-        __IPYTHON__  # type: ignore
+        eval('__IPYTHON__')
```

Comment:
looks good, maybe just worth testing this actually works.

Reply:
It works for me in ipython (no error), and standard python (raises NameError as expected). Let me know if you want a test added.